File: ios/Classes/WebRTCService.m
#import "WebRTCService.h"

@implementation WebRTCService

// Static variable for the singleton instance
static WebRTCService *instance = nil;

// Private initializer to prevent instantiation from outside
- (instancetype)initPrivate {
    self = [super init];
    if (self) {
        // Initialization logic if any
    }
    return self;
}

// Singleton instance method
+ (instancetype)sharedInstance {
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        instance = [[self alloc] initPrivate];
    });
    return instance;
}

// Method to set the Processor
- (void)setVideoProcessor:(VideoProcessor *)videoProcessor {
    _videoProcessor = videoProcessor;
}

// Method to get the current Processor
- (VideoProcessor *)getVideoProcessor {
    return _videoProcessor;
}

@end

File: android/src/main/java/live/videosdk/webrtc/WebRTCService.java
package live.videosdk.webrtc;

import live.videosdk.webrtc.VideoProcessor;
import android.util.Log;

public class WebRTCService {

    private static final String TAG = "WebRTCService";
    private static WebRTCService instance;
    
    private VideoProcessor videoProcessor;

    // Private constructor to prevent instantiation from outside
    private WebRTCService() {
        // Initialization logic if any
    }

    // Static method to get the singleton instance
    public static synchronized WebRTCService getInstance() {
        if (instance == null) {
            instance = new WebRTCService();
        }
        return instance;
    }

    // Method to set the VideoProcessor
    public void setVideoProcessor(VideoProcessor videoProcessor) {
        this.videoProcessor = videoProcessor;
    }

    // Method to get the current VideoProcessor
    public VideoProcessor getVideoProcessor() {
        return videoProcessor;
    }

    // Other methods related to WebRTC service can be added here
}
